pkg/controller/reconciler/deploymenttemplate_reconciler.go

```diff
@@ -0,0 +1,546 @@
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to poll operation status: %w", err)
+		}
+
+		if !poller.Done() {
+			return ctrl.Result{Requeue: true, RequeueAfter: r.requeueDelay()}, nil
+		}
+
+		// If we get here, the operation is complete.
+		resp, err := poller.Result(ctx)
+		if err != nil {
+			// Operation failed, reset state and retry.
+			r.EventRecorder.Event(deploymentTemplate, corev1.EventTypeWarning, "ResourceError", err.Error())
+			logger.Error(err, "Update failed.")
+
+			deploymentTemplate.Status.Operation = nil
+			deploymentTemplate.Status.Phrase = radappiov1alpha3.DeploymentTemplatePhraseFailed
+			deploymentTemplate.Status.Message = err.Error()
+			err = r.Client.Status().Update(ctx, deploymentTemplate)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			return ctrl.Result{Requeue: true, RequeueAfter: r.requeueDelay()}, nil
+		}
+
+		logger.Info("Creating output resources.")
+
+		// Get outputResources from the response
+		outputResources := make([]string, 0)
+		if resp.Properties["outputResources"] != nil {
+			outputResourceList := resp.Properties["outputResources"].([]any)
+			for _, resource := range outputResourceList {
+				outputResource := resource.(map[string]any)
+				outputResources = append(outputResources, outputResource["id"].(string))
+			}
+
+			// Compare outputResources with existing DeploymentResources
+			// if is present in deploymentTemplate.Status.OutputResources but not in outputResources, delete it
+			// if is not present in deploymentTemplate.Status.OutputResources but is in outputResources, create it
+			// if is present in both, do nothing
+
+			existingOutputResources := make(map[string]bool)
+			for _, resource := range deploymentTemplate.Status.OutputResources {
+				existingOutputResources[resource] = true
+			}
+
+			newOutputResources := make(map[string]bool)
+			for _, resource := range outputResources {
+				newOutputResources[resource] = true
+			}
+
+			for _, outputResourceId := range outputResources {
+				if _, ok := existingOutputResources[outputResourceId]; !ok {
+					// Resource is not present in deploymentTemplate.Status.OutputResources but is in outputResources, create it
+
+					resourceName := generateDeploymentResourceName(outputResourceId)
+					deploymentResource := &radappiov1alpha3.DeploymentResource{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      resourceName,
+							Namespace: deploymentTemplate.Namespace,
+						},
+						Spec: radappiov1alpha3.DeploymentResourceSpec{
+							Id:             outputResourceId,
+							ProviderConfig: deploymentTemplate.Spec.ProviderConfig,
+							RootFileName:   deploymentTemplate.Spec.RootFileName,
+						},
+					}
+
+					if controllerutil.AddFinalizer(deploymentResource, DeploymentResourceFinalizer) {
+						// Add the DeploymentTemplate as the owner of the DeploymentResource
+						if err := controllerutil.SetControllerReference(deploymentTemplate, deploymentResource, r.Scheme); err != nil {
```

Comment:
I see you've already got an owner reference here. We should be using that for lookups instead of the `rootFileName`.

Reply:
fixed